logger/src/config.rs

```diff
@@ -0,0 +1,99 @@
+//! Module containing logic related to spawning a logger from the
+//! configuration, as well as run-time reloading of the log-level.
+
+use iroha_config::{
+    derive::Configurable,
+    runtime_upgrades::{ReloadFailure, ReloadHandle, Reloadable},
+};
+use serde::{Deserialize, Serialize};
+use tracing::Subscriber;
+use tracing_subscriber::{filter::LevelFilter, reload::Handle};
+
+const DEFAULT_MAX_LOG_LEVEL: LevelEnv = LevelEnv::DEBUG;
+
+use super::{debug, Level};
+
+/// Log level for reading from environment and (de)serializing
+#[allow(clippy::upper_case_acronyms)]
+#[derive(Debug, Deserialize, Serialize, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub enum LevelEnv {
+    /// Error
+    ERROR,
+    /// Warn
+    WARN,
+    /// Info
+    INFO,
+    /// Debug
+    DEBUG,
+    /// Trace
+    TRACE,
+}
+
+impl Default for LevelEnv {
+    fn default() -> Self {
+        DEFAULT_MAX_LOG_LEVEL
+    }
+}
+
+impl From<LevelEnv> for Level {
+    fn from(level: LevelEnv) -> Self {
+        match level {
+            LevelEnv::ERROR => Self::ERROR,
+            LevelEnv::TRACE => Self::TRACE,
+            LevelEnv::INFO => Self::INFO,
+            LevelEnv::DEBUG => Self::DEBUG,
+            LevelEnv::WARN => Self::WARN,
+        }
+    }
+}
+
+impl From<LevelEnv> for LevelFilter {
+    fn from(level: LevelEnv) -> Self {
+        Level::from(level).into()
+    }
+}
+
+impl<T: Subscriber> Reloadable<LevelEnv> for Handle<LevelFilter, T> {
+    fn reload(
+        &mut self,
+        level: LevelEnv,
+    ) -> Result<(), iroha_config::runtime_upgrades::ReloadFailure> {
+        debug!("Reload Logging level");
+        let level_filter = tracing_subscriber::filter::LevelFilter::from_level(level.into());
+        Handle::reload(self, level_filter).map_err(|_todo| ReloadFailure::Other)
+    }
+}
+
+/// Configuration for `Logger`.
+#[derive(Clone, Deserialize, Serialize, Debug, Configurable)]
+#[serde(rename_all = "UPPERCASE")]
+#[serde(default)]
+pub struct Configuration {
+    /// Maximum log level
+    #[config(serde_as_str)]
+    pub max_log_level: LevelEnv,
+    /// Capacity (or batch size) for telemetry channel
+    pub telemetry_capacity: usize,
+    /// Compact mode (no spans from telemetry)
+    pub compact_mode: bool,
+    /// If provided, logs will be copied to said file in the
+    /// format readable by [bunyan](https://lib.rs/crates/bunyan)
+    pub log_file_path: Option<std::path::PathBuf>,
+    /// Internal: handle to reload the log level.
+    pub log_reload_handle: ReloadHandle<LevelEnv>,
```

Comment:
Is there a need for whole mutex here? We can use just atomic for that (or `crossbeam::AtomicCell`)